tests/integration-tests/tests/networking/test_security_groups.py

```diff
@@ -85,27 +86,67 @@ def test_overwrite_sg(region, custom_security_group, pcluster_config_reader, clu
         assert_that(mount_target_security_groups[0]).is_equal_to(custom_security_group_id)
         assert_that(mount_target_security_groups).is_length(1)
 
+    if scheduler == "slurm":
+        _check_connections_between_head_node_and_compute_nodes(cluster)
+        updated_config_file = pcluster_config_reader(
+            config_file="pcluster.config.update.yaml",
+            vpc_security_group_id=custom_security_group_id,
+        )
+        cluster.update(str(updated_config_file), force_update="true")
```

Comment:
let'add a comment here describing we're modifying SG for both head node and compute and why this update is useful